packages/angular_devkit/core/src/jobs/README.md

```diff
@@ -0,0 +1,464 @@
+
+# Creating Jobs
+
+A job is at its core a function with a Schema associated to it;
+
+```typescript
+import { Observable } from 'rxjs';
+import { jobs } from '@angular-devkit/core';
+
+const argument = {
+  type: 'array', items: { type: 'number' },
+};
+const output = {
+  type: 'number',
+};
+
+export function add(argument: number[]): Observable<jobs.JobEvent<number>> {
+  return new Observable(o => {
+    o.next({
+      kind: jobs.JobEventKind.Output,
+      output: argument.reduce((total, curr) => total + curr, 0),
+    });
+    o.complete();
+  });
+}
+
+// Add a property to `add` to make it officially a JobHandler. The Job system does not recognize any
+// function as a JobHandler.
+add.jobDescription = {
+  argument: argument,
+  output: output,
+};
+
+// Call the job with an array as argument, and log its output.
+declare const scheduler: jobs.Scheduler;
+scheduler.schedule('add', [1, 2, 3, 4])
+    .output.subscribe(x => console.log(x));  // Will output 10.
+```
+
+This is a lot of boilerplate, so we made some helpers to improve readability and manage argument, 
+input and output automatically:
+
+```typescript
+// Add is a JobHandler function, like the above.
+export const add = jobs.createJob<number[], number>(
+  argument => argument.reduce((total, curr) => total + curr, 0),
+);
+
+// Schedule like above.
+```
+
+You can also return a Promise or an Observable, as jobs are asynchronous. This helper will set 
+progress to 0, start and end events appropriately, and will pass in a logger. It will also manage
+channels automatically (see below).
+
+A more complex job can be declared like this:
+
+```typescript
+import { Observable } from 'rxjs';
+import { JobRegistry } from './api'
+
+// Show progress with each count. Output "more" in a channel.
+export const count = jobs.createJob<number, number>(
+  // Receive a context that contains additional methods, like progress, logger and channels.
+  (argument: number, { progress, channels }) => new Observable<number>(o => {
+    let i = 0;
+    function doCount() {
+      o.next(i++);
+      progress(i / argument);
+      channels['side'].next('more');
+  
+      if (i < argument) {
+        setTimeout(doCount, 100);
+      } else {
+        o.complete();
+      }
+    }
+    setTimeout(doCount, 100);
+  }),
+  {
+    argument: { type: 'number' },
+    output: { type: 'number' },
+    channels: {
+      'side': { type: 'string', const: 'more' },
+    },
+  },
+);
+
+// Get a hold of a scheduler that refers to the job above.
+declare const scheduler: jobs.Scheduler;
+
+const job = scheduler.schedule('count', 0);
+job.channels['side'] && job.channels['side'].subscribe(x => console.log(x));
+const c = job.getChannel<string>('side', { type: 'string' });
+if (c) {
+  c.subscribe((x: string) => console.log(x));
+}
+```
+
+## <a name="Communicating"></a>Communicating With Jobs
+Jobs can be started and updated in a separate process or thread, and as such communication with a 
```

Comment:
A good inter-process communication strategy is also of use in the current forked type checker in ngtools/webpack.

Reply:
I saw that, and it was considered as part of the requirements. I'd like to have it move to Jobs (I think it'd be fairly easy).